src/rust/ide/src/controller/module.rs

```diff
@@ -140,6 +142,12 @@ shared! { Handle
             self.module.ast.repr()
         }
 
+        /// Obtains definition information for given graph id.
+        pub fn find_definition(&self,id:&dr::graph::Id) -> FallibleResult<DefinitionInfo> {
+            let module = known::Module::try_new(self.module.ast.clone())?;
```

Comment:
This is a good opportunity to make `ast` field in module controller be of type `known::Module`, and make this try_new once after reading file. Probably all tests should be fixed (there are tests loading simple 2+2 as a module file)